### ISSUE TITLE
make preboot smaller

### ISSUE THREAD

**Issue by jeffwhelpley** [angular/universal#257]
Currently preboot is ~10k (depending on which options are chosen). We should be able to get this down to ~2K with some effort.

**Comment by jeffwhelpley** [comment]
btw, I can work with whoever wants to take this one. There are some clear quick wins, but some stuff that will be tricky.

**Comment by jeffwhelpley** [comment]
Last night as my brain was catching up all the craziness from ng-conf I realized how we are going to drastically reduce the size of preboot. It basically comes down to two things. I always thought we would gain a lot by implementing webpack 2 with tree shaking so that only the code for the options that are chosen are included. That will help, but not nearly as much as simply chopping off all the replay logic. What I realized is that the inline JavaScript included in the initial page load really only needs logic for recording events and perhaps freeze (if that option is chosen). Everything else can be included as a separate client side library that the user adds to their larger client side package (i.e. along with angular itself and their custom app code). I am fairly certain this will get the size of the inline code down to 2K or less.
